app/exporters/shopify/product_exporter.rb

```diff
@@ -0,0 +1,90 @@
+module Shopify
+  class ProductExporter
+    def initialize(spree_product_id, factory = nil, logger = nil)
+      @logger = logger || default_logger
+      @factory = factory || default_factory
+      @spree_product = Spree::Product.find(spree_product_id)
+      @original_shopify_product = find_or_initialize(spree_product)
+    end
+
+    def perform
+      # NOTE(cab): Refactor the new.perform to use call?
+      shopify_product = factory.new(spree_product, original_shopify_product).perform
```

Comment:
If we decorate the `ActiveResource::Base` class with something like that:

https://github.com/glossier/solidus_retail/blob/8bf611f0ee385d556958ff2ea2dd656fbf924544/lib/active_resource/base_decorator.rb

This would remove the need of having to have a factory class that does exactly this.